File: core/models/desk.py
from django.db import models


class Desk(models.Model):
    floor = models.ForeignKey('Floor', on_delete=models.CASCADE, related_name='desks')

    name = models.CharField(max_length=32)
    description = models.TextField(blank=True, default='')
    tags = models.JSONField(default=list)

    width = models.FloatField()
    height = models.FloatField()
    pos_x = models.FloatField()
    pos_y = models.FloatField()
    orientation = models.FloatField()

File: app/src/models/desk.ts
export interface Desk {
    id: number;
    name: string;
    description: string;
    tags: string[];

    width: number;
    height: number;
    pos_x: number;
    pos_y: number;
    orientation: number;

    floor: number;
}

export interface DisplayDesk extends Desk {
    is_blocked: boolean;
}

export function isDisplayDesk(obj: any): obj is DisplayDesk {
    return obj != null && obj.is_blocked != null;
}